src/Core/Form/IdentifiableObject/Builder/FormBuilder.php

```diff
@@ -98,6 +111,9 @@ public function getForm(array $data = [], array $options = [])
     public function getFormFor($id, array $data = [], array $options = [])
     {
         $data = array_merge($this->dataProvider->getData($id), $data);
+        if (null !== $this->optionsProvider) {
+            $options = array_merge($this->optionsProvider->getOptions($id, $data), $options);
+        }
```

Comment:
```suggestion
        if (null !== $this->optionsProvider
            && is_array($providerOptions = $this->optionsProvider->getOptions($id, $data))
        ) {
            $options = array_merge($providerOptions, $options);
        }
```

The interface can return other thing than an array,

Reply:
Oh right, and unlike the data provider, the options is always an array So I changed the typing and force `array`